drivers/src/pci/bus.rs

```diff
@@ -45,12 +45,50 @@ impl PciBus {
             }
         }
 
-        // TODO: Recursively enumerate any bridges on this bus.
+        // Recursively enumerate the buses behind any bridges on this bus.
+        let mut cur_bus = bus_num;
+        for bd in devices.iter() {
+            let bridge_id = bd.1;
+            let sec_bus = cur_bus.next().ok_or(Error::OutOfBuses)?;
+            // Unwrap ok, we just added the device above.
+            let pci_dev = device_arena.get_mut(bridge_id).unwrap();
+            match pci_dev {
+                PciDeviceType::Bridge(bridge) => {
+                    // Set the bridge to cover everything beyond sec_bus until we've enumerated
+                    // the buses behind the bridge.
+                    bridge.assign_bus_range(BusRange {
+                        start: sec_bus,
+                        end: Bus::max(),
+                    });
+                }
+                _ => continue,
+            };
+
+            let child_bus = PciBus::enumerate(config_space, sec_bus, device_arena)?;
+            let sub_bus = child_bus.subordinate_bus_num();
+
+            // The arena may have re-allocated; re-acquire the reference to the bridge device.
```

Comment:
doesn't get_mut mutably borrow the arena? Does that prevent it from re-allocating?